src/creds.rs

```diff
@@ -0,0 +1,555 @@
+pub fn verifiable_credential_backup_encryption_key_hex(
+    seed_hex: String,
+    net: String,
+) -> Result<String, ConcordiumWalletCryptoError> {
+    get_verifiable_credential_backup_encryption_key_aux(seed_hex, net.as_str()).map_err(|e| {
+        ConcordiumWalletCryptoError::CallFailed {
+            call: format!("verifiable_credential_backup_encryption_key_hex(seed_hex, net={net}"),
+            msg: format!("{:#}", e),
+        }
+    })
+}
+
+/// UniFFI compatible bridge to [`IdentityObjectRequestInput`],
+/// providing the implementation of the UDL declaration of the same name.
+/// The translation is performed using Serde.
+#[derive(Clone, Debug, Serialize)]
+pub struct IdentityIssuanceRequestParameters {
+    #[serde(rename = "ipInfo")]
+    pub ip_info: IdentityProviderInfo,
+    #[serde(rename = "globalContext")]
+    pub global_context: GlobalContext,
+    #[serde(rename = "arsInfos")]
+    pub ars_infos: HashMap<u32, AnonymityRevokerInfo>,
+    #[serde(rename = "arThreshold")]
+    pub ar_threshold: u8,
+    #[serde(rename = "prfKey")]
+    pub prf_key_hex: String,
+    #[serde(rename = "idCredSec")]
+    pub id_cred_sec_hex: String,
+    #[serde(rename = "blindingRandomness")]
+    pub blinding_randomness_hex: String,
+}
+
+/// UniFFI compatible bridge to [`IdentityRecoveryRequestInput`],
+/// providing the implementation of the UDL declaration of the same name.
+/// The translation is performed using Serde.
+#[derive(Clone, Debug, Serialize)]
+pub struct IdentityRecoveryRequestParameters {
+    #[serde(rename = "ipInfo")]
+    pub ip_info: IdentityProviderInfo,
+    #[serde(rename = "globalContext")]
+    pub global_context: GlobalContext,
+    #[serde(rename = "timestamp")]
+    pub timestamp: u64,
+    #[serde(rename = "idCredSec")]
+    pub id_cred_sec_hex: String,
+}
+
+/// UniFFI compatible bridge to [`concordium_base::id::types::IpInfo<concordium_base::id::constants::IpPairing>`],
+/// providing the implementation of the UDL declaration of the same name.
+/// The translation is performed using Serde.
+#[derive(Clone, Debug, Serialize)]
+pub struct IdentityProviderInfo {
+    #[serde(rename = "ipIdentity")]
+    pub identity: u32,
+    #[serde(rename = "ipDescription")]
+    pub description: Description,
+    #[serde(rename = "ipVerifyKey")]
+    pub verify_key_hex: String,
+    #[serde(rename = "ipCdiVerifyKey")]
+    pub cdi_verify_key_hex: String,
+}
+
+impl TryFrom<GlobalContext> for concordium_base::id::types::GlobalContext<ArCurve> {
+    type Error = uniffi::deps::anyhow::Error;
+
+    fn try_from(value: GlobalContext) -> Result<Self, Self::Error> {
+        serde_json::to_string(&value)
+            .context("cannot encode request object as JSON")
+            .and_then(|json| {
+                serde_json::from_str::<concordium_base::id::types::GlobalContext<ArCurve>>(&json)
+                    .context("cannot decode request object into internal type")
+            })
+    }
+}
```

Comment:
can we move the `GlobalContext` in this file, alternatively can this be moved into `types.rs` where the `GlobalContext` is defined.